storage/appinsights/appinsights.go

```diff
@@ -0,0 +1,109 @@
+package appinsights
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/microsoft/ApplicationInsights-Go/appinsights"
+	"github.com/sourcegraph/checkup/types"
+)
+
+// Type should match the package name
+const Type = "appinsights"
+
+// Storage will send results to Azure Application Insights
+type Storage struct {
+	// InstrumentationKey is a GUID that identifies an app insights instance
+	InstrumentationKey string `json:"instrumentation_key"`
+
+	// TestName identifies the test name sent
+	// in Application Insights trackAvailability() events
+	TestName string `json:"test_name,omitempty"`
+
+	// TestLocation identifies the test location sent
+	// in Application Insights trackAvailability() events
+	TestLocation string `json:"test_location,omitempty"`
+
+	// Tags will be applied to all telemetry items and
+	// visible in the customProperties field when viewing the
+	// submitted data
+	Tags map[string]string `json:"tags,omitempty"`
+
+	// TelemetryClient is the appinsights.Client with which to
+	// send Application Insights trackAvailability() events
+	// Automatically created if InstrumentationKey is set.
+	TelemetryClient appinsights.TelemetryClient
```

Comment:
This shouldn't be a public property, prefix also stutters, use `client appinsights.TelemetryClient` (like github does).